btm/src/test/java/bitronix/tm/mock/JdbcSharedConnectionTest.java

```diff
@@ -87,6 +87,63 @@ public void run() {
 
     }
 
+    public void testSharedConnectionQuickSuspendEnabledResumeOnAnotherThread() throws Exception {
+       if (log.isDebugEnabled()) { log.debug("*** Starting testSharedConnectionQuickSuspendEnabledResumeOnAnotherThread: getting TM"); }
+       TransactionManagerServices.getConfiguration().setQuickSuspend(true); 
+       final BitronixTransactionManager tm = TransactionManagerServices.getTransactionManager();
+       tm.setTransactionTimeout(120);
+
+       if (log.isDebugEnabled()) { log.debug("*** before begin"); }
+       tm.begin();
+       if (log.isDebugEnabled()) { log.debug("*** after begin"); }
+
+       final Transaction suspended = tm.suspend();
+
+       final ArrayList<Connection> twoConnections = new ArrayList<Connection>();
+       Thread thread1 = new Thread() {
+           @Override
+       	public void run() {
+       		try {
+					tm.resume(suspended);
+			        if (log.isDebugEnabled()) { log.debug("*** getting connection from DS1"); }
+			        Connection connection = poolingDataSource1.getConnection();
+			        connection.createStatement();
+			        twoConnections.add(connection);
+			        tm.suspend();
```

Comment:
A functional difference here.  On the non-quick version of this test we are able to call tm.resume(s..d) from both threads without suspending it on the first thread.  In theory that looks like resume() can be used to make the transaction active on multiple threads.  

I cannot tell if that is intentional behavior, or a side-effect of iterations of the suspend/resume approach.  

When i read through the older code (where TMSUSPEND is passed as a flag), it seems that calling resume from multiple threads (without suspend on the first), would fail with the same exception that quick-suspend now does.

I believe this is the safest behavior, because i cannot reason about how a transaction that is active on multiple threads would work in the majority of code i've read through.  But i'm willing to reconsider if you think its a valuable feature.